sql/core/src/test/resources/sql-tests/inputs/comparator.sql

```diff
@@ -1,3 +1,12 @@
 -- binary type
 select x'00' < x'0f';
 select x'00' < x'ff';
+
+-- trim string to numeric
+select '1 ' = 1Y;
+select '1 ' = 1S;
+select '1 ' = 1;
```

Comment:
For this simple case, how much slower we are compared to the master branch?

Reply:
let me test this.